File: fixtures/swift/inherits.swift
protocol Describe {
    func describe() -> String
}

class Base {
    func area() -> Int {
        return 1
    }
}

class Circle: Base, Describe {
    func describe() -> String {
        return "circle"
    }
}

struct Plain {
    func value() -> Int {
        return 1
    }
}

class Container<T> {
    func first() -> T? {
        return nil
    }
}

class Wide: Container<Int> {
    func widen() -> Int {
        return 2
    }
}

extension Circle: Swift.Equatable {}

File: fixtures/javascript/inherits.js
class Describe {
  describe() {
    return "thing";
  }
}

class Circle extends Describe {
  describe() {
    return "circle";
  }
}

class Plain {
  value() {
    return 1;
  }
}

class Container {}

const ns = { Container };

class Wide extends ns.Container {
  value() {
    return 1;
  }
}
